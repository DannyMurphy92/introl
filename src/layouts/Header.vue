<template>
  <div class="site-trunk header">
    <Logo class="header__logo" />
    <Accordion>
      <template #header>
        <span>Click meeee</span>
      </template>
      <template #body>
        <span>Hello</span>
      </template>
    </Accordion>
  </div>
</template>
<script lang="ts">
import Vue from 'vue';
import Logo from '@/components/Logo.vue';
import Accordion from '@/components/Accordion.vue';

export default Vue.extend({
  name: 'Header',
  components: {
    Logo,
    Accordion,
  },
});
</script>

<style lang="scss" scoped>
.header__logo {
  width: 200px;
  height: 100px;
}
</style>
