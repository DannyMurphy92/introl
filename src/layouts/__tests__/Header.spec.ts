import Header from '../Header.vue';
import { Wrapper, shallowMount } from '@vue/test-utils';

describe('Header', () => {
  let wrapper: Wrapper<any>;

  const mountComponent = () => {
    wrapper = shallowMount(Header);
  };

  beforeEach(() => {
    mountComponent();
  });

  it('THEN renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('THEN renders the logo', () => {
    expect(wrapper.findComponent({ name: 'Logo' }).exists()).toBe(true);
  });
});
