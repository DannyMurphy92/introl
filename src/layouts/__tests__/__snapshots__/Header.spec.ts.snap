// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header THEN renders correctly 1`] = `
<div class="header">
  <div class="site-trunk header__container">
    <div class="header__logo__container"><img src="@/static/logo.png" alt="logo" class="header__logo"></div>
    <div>
      <accordion-stub class="header__contact"></accordion-stub>
    </div>
  </div>
</div>
`;
